### PR TITLE
Fix issue 1703 - Adding ExpandCollapse pattern to DataGridViewComboBoxCell

### LINKED ISSUE #1703: Accessibility: DataGridView cell of type ComboBox must support the ExpandCollapse pattern
DataGridView ComboBoxCell has ComboBox control type. UI elements which have ComboBox control type in corresponding accessibility property should provide ExpandCollapse pattern and all related functionality.

DevDiv issue: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/956708

### PR DESCRIPTION
**Pull request by M-Lipin**
Fixes #1703

## Proposed changes

- Adding ExpandCollapsePatternProvider to DataGridViewComboBox cell as corresponding cell has ComboBox control type. So adding pattern is obligatory.
- ExpandCollapsePattern is also presented in child ComboBox control: made the value of ExpandCollapseState synced between ComboBox and DataGridViewComboBoxCell.

## Customer Impact

- DataGridView UI will be more accessible for users with visual impairment. 
- Users will be notified about the control state during DataGridView cell navigation without necessity to expand/collapse the ComboBox.

## Regression? 

- No, but conditionally is dependent on the changes with DataGridView (.NET Framework 4.7.2)

## Risk

- Lowest.

### Before

image

### After

image

## Test methodology 

- Manual testing, accessibility validation tools.
- Waiting for test results from QA.
- Automated tests and apps.

## Accessibility testing  

The change does not modify current UI itself, but only fixes the accessibility issue.
 

## Test environment(s) 

- 
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview8-012929
 Commit:    795f8aeaa8

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview8-012929\

Host (useful for support):
  Version: 3.0.0-preview8-27907-09
  Commit:  fc924dc319

.NET Core SDKs installed:
  2.1.700-preview-009601 [C:\Program Files\dotnet\sdk]
  2.1.800-preview-009696 [C:\Program Files\dotnet\sdk]
  3.0.100-preview8-012929 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview8.19351.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview8-27907-09 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview8-27907-09 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1704)

### CODE REVIEW

**Comment by M-Lipin** [comment]
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/956708

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxCell.cs [hunk 1]
```diff
@@ -3119,9 +3119,37 @@ internal override object GetPropertyValue(int propertyID)
                 {
                     return NativeMethods.UIA_ComboBoxControlTypeId;
                 }
+                else if (propertyID == NativeMethods.UIA_IsExpandCollapsePatternAvailablePropertyId)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxCell.cs]: `else` is redundant

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxCell.cs]: can we use `switch`/pattern matching here?

**Comment by M-Lipin** [comment]
QA passed.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxCell.cs (hunk 1 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxCell.cs]: Fixed this. Thanks!

**Comment by zsd4yr** [comment]
Approved

**Pull request merged**